packages/plugins/src/hardhat/simulate.ts

```diff
@@ -249,6 +258,40 @@ export const setupPresimulationState = async (
   return signer
 }
 
+/**
+ * Fetches the transaction response objects for all of the transactions in the simulation. There's some additional
+ * data in the responses that is not in the receipts (data, value, etc) which is not available in the receipts. We
+ * use this information to calculate deployment cost estimates on the website.
+ */
+export const fetchTransactionResponses = async (
+  receipts: Array<SphinxTransactionReceipt>,
+  provider: HardhatEthersProvider
+): Promise<SimulationTransactions> => {
+  // Since the size of receipts array is unbounded, we use pLimit to reduce the number of simultaneous calls.
+  // This reduces the chance of us triggering a rate limit in the RPC provider.
+  const transactions = await Promise.all(
+    receipts.map(async (receipt) => {
+      const response = await limit(async () =>
+        provider.getTransaction(receipt.hash)
+      )
```

Comment:
See the comment, we should probably try to use plimit whenever we have situations where we're using `Promise.all()` over a potentially large number of rpc provider calls to try to avoid getting rate limited if possible.